bundles/org.openhab.core.config.discovery.addon/src/main/java/org/openhab/core/config/discovery/addon/AddonSuggestionService.java

```diff
@@ -173,31 +173,43 @@ public void removeAddonInfoProvider(AddonInfoProvider addonInfoProvider) {
 
     @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
     public void addAddonFinder(AddonFinder addonFinder) {
-        addonFinders.add(addonFinder);
+        synchronized (addonFinders) {
+            addonFinders.add(addonFinder);
+        }
         changed();
     }
 
     public void removeAddonFinder(AddonFinder addonFinder) {
-        if (addonFinders.remove(addonFinder)) {
+        boolean removed;
+        synchronized (addonFinders) {
+            removed = addonFinders.remove(addonFinder);
+        }
+        if (removed) {
```

Comment:
Unrelated to this PR since logic is unchanged: But why is it needed to call this method after removing a finder? It seems the method will only iterate the remaining finders (presumably again?), but the removed finder is unaffected by this?

Reply:
Fixed